test/e2e/multinamespace/multinamespace_test.go

```diff
@@ -0,0 +1,142 @@
+package multinamespace
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	ispnv1 "github.com/infinispan/infinispan-operator/pkg/apis/infinispan/v1"
+	cconsts "github.com/infinispan/infinispan-operator/pkg/controller/constants"
+	ispnctrl "github.com/infinispan/infinispan-operator/pkg/controller/infinispan"
+	ispn "github.com/infinispan/infinispan-operator/pkg/infinispan"
+	tconst "github.com/infinispan/infinispan-operator/test/e2e/constants"
+	k8s "github.com/infinispan/infinispan-operator/test/e2e/k8s"
+	tutils "github.com/infinispan/infinispan-operator/test/e2e/utils"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var testKube = k8s.NewTestKubernetes(os.Getenv("TESTING_CONTEXT"))
+var serviceAccountKube = k8s.NewTestKubernetes("")
+
+var log = logf.Log.WithName("main_test")
+
+var MinimalSpec = ispnv1.Infinispan{
+	TypeMeta: tconst.InfinispanTypeMeta,
+	ObjectMeta: metav1.ObjectMeta{
+		Name: tconst.DefaultClusterName,
+	},
+	Spec: ispnv1.InfinispanSpec{
+		Replicas: 1,
+	},
+}
+
+func TestMain(m *testing.M) {
+	nsAsString := strings.ToLower(tconst.MultiNamespace)
+	namespaces := strings.Split(nsAsString, ",")
+	if "TRUE" == tconst.RunLocalOperator {
+		for _, namespace := range namespaces {
+			testKube.DeleteNamespace(namespace)
+		}
+		testKube.DeleteCRD("infinispans.infinispan.org")
+		testKube.DeleteCRD("caches.infinispan.org")
+		testKube.DeleteCRD("backup.infinispan.org")
+		testKube.DeleteCRD("restore.infinispan.org")
+		for _, namespace := range namespaces {
```

Comment:
Can't we do the deletion and creation of namespaces in the same loop?

Reply:
I just wanted to keep cleanup separated from setup.